meso_tools/NAS_tools.py

```diff
@@ -0,0 +1,96 @@
+        This data is the result of an AllenSDK query and not currently included in this code
+    """    
+
+    user = credentials['SYNOLOGY_USERNAME']
+    password = credentials['SYNOLOGY_PASSWORD']
+    ip = credentials['SYNOLOGY_IP']
+
+    # login
+    response = requests.get("http://"+str(ip)+"/webapi/auth.cgi?api=SYNO.API.Auth&version=2&method=login&account="+str(user)+ 
+        "&passwd="+str(password)+"&session=FileStation&format=sid")
+    sid = (response.json()['data']['sid'])
+    if nas_name == 'ophys_nas':
+        #folder 1
+        response = requests.get("http://"+str(ip)+"/webapi/entry.cgi?api=SYNO.FileStation.List&version=2&method=list&additional=%5B%22real_path%22%2C%22size%2Cperm%22%5D&folder_path=%22%2Fmeso2_backup/Data_Backup%22&_sid="+sid)        
+        released_backups = release_check(release_list, response)
+        
+        #folder 2
+        response = requests.get("http://"+str(ip)+"/webapi/entry.cgi?api=SYNO.FileStation.List&version=2&method=list&additional=%5B%22real_path%22%2C%22size%2Cperm%22%5D&folder_path=%22%2Fmeso1_backup/Data_Backup%22&_sid="+sid)
+        released_backups2 = release_check(release_list, response)
+    elif nas_name == 'ophys_backup':
+
+        #folder 1
+        response = requests.get("http://"+str(ip)+"/webapi/entry.cgi?api=SYNO.FileStation.List&version=2&method=list&additional=%5B%22real_path%22%2C%22size%2Cperm%22%5D&folder_path=%22%2Fmeso_backup_full/data_backup%22&_sid="+sid)
+        released_backups = release_check(release_list, response)
+        #folder 2
+        response = requests.get("http://"+str(ip)+"/webapi/entry.cgi?api=SYNO.FileStation.List&version=2&method=list&additional=%5B%22real_path%22%2C%22size%2Cperm%22%5D&folder_path=%22%2Fmeso_backup/data_backup%22&_sid="+sid)
+        released_backups2 = release_check(release_list, response)          
+    else:
+        print('NAS name not valid')
+        released_backups = []
+        released_backups2 = []
+    # logout
+
+    requests.get("http://"+str(ip)+"/webapi/auth.cgi?api=SYNO.API.Auth&version=6&method=logout&session=FileStation")
+    return(released_backups, released_backups2)
+            
+def release_check(release_list: str, query_response: requests.models.Response):
+    """Compares the output of a requests query against a list of released sessions, 
+    and returns the filepath for each session in the NAS storage that matches the ID of a released session
+
+    Args:
+        release_list (str): Relative string filepath to a .csv file containing information about released ophys sessions. 
+        This data is the result of an AllenSDK query and not currently included in this code
+        query_response (requests.models.Response): Response from the NAS storage containing list of folders and filepaths.  
+        This data is the output of the nas_query function
+    """    
+    import pandas as pd
```

Comment:
do this in the import section at the top please

Reply:
done